### ISSUE TITLE
Unable to setup OIDC with a domain

### ISSUE THREAD

**Issue by barlock** [kubeflow/kubeflow#4400]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I'm trying to deploy an authenticated Kubeflow deployment onto an existing k8s cluster on IBM's k8s service.

I'm using the `existing_arikto` config. After generating and applying (or just applying in the 0.7 branch) I'm able to access the working OIDC with IP addresses.

After an unknown period of time, I can no longer access it with the IP address. 

For the 0.7 branch I use IKS's managed Istio and Knative "plugin" to configure both for IBM's ingresses. 

Once deployed, I change the `host` in the `kubeflow-gateway` from `*` to my domain, something like (kubeflow.<cluster_name>.<region>.containers.appdomain.cloud), for both kubeflow and dex in the 0.6.2 branch, and for just kubeflow in the 0.7 branch. I can't access the kubeflow UI with or without Auth, It gives a `404` for the dashboard and a 504 for dex. 

I also tried leaving the gateway as `*` and changing the virtual services's `host`, same result.

**What did you expect to happen:**
I expected to have a kubeflow deployment accessible with auth through dex at the domain specified. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
* I've looked through the code and couldn't find any place to specify a hostname, it seems that the manifests hardcode a `*`. Is there a supported way to configure a domain? The docs hint at configuring at it and forcing https, both sound great, I'm not sure how to make it work though.
* If this is possible, docs should probably be updated, I found a `KUBEFLOW_ENDPOINT` and `OIDC_ENDPOINT` those change the values in the services, but not in the router.
* I _think_ IBM's ingress terminates TLS before it gets into the cluster, but I'm uncertain, I do have certificates inside the cluster I could configure, but I don't know what to do with them, `istio-ingressgatway-certs` in the `istio-system` namespace seems promising, but I don't know.
* A base deploy from current master, with istio being installed by an addon, and not via kfctl, returns `504`s for both port forwarding and using the loadbalancer's IP

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): 0.6.2 and 0.7.0-rc5
- kfctl version: (use `kfctl version`): 0.6.2 and 0.7.0-rc5
- Kubernetes platform: (e.g. `minikube`) IBM IKS
- Kubernetes version: (use `kubectl version`): 1.15.5
- OS (e.g. from `/etc/os-release`): Ubuntu
- Istio Version (deployed manually with addon): 1.3.3

**Comment by barlock** [comment]
I'm still uncertain about the 0.6.x branch, but I've been able to get it working for the most part in the 0.7-rc5 branch. Though I am having issues configuring, but I'll open a new issue...

The one thing I had to do was I would get `RBAC access denied` (similar issue https://github.com/kubeflow/kubeflow/issues/3975) and if I turn `OFF` the `clusterRbacConfig` it goes away. 

That seems fine, but I'm uncertain what effect it has generally on the rest of the system.
